src/gt4py/backend/templates/stencil_module.py.in

```diff
@@ -115,8 +115,16 @@ class {{ class_name }}(StencilObject):
     def __call__(
         self, {{ stencil_signature }}, domain=None, origin=None, validate_args=True, exec_info=None
     ):
-        if exec_info is not None:
-            exec_info["call_start_time"] = time.perf_counter()
+        stencil_info = (
+            exec_info.get("{{ class_name }}", {})
+            if exec_info is not None else None
+        )
+        if stencil_info is not None:
+            exec_info["{{ class_name  }}"] = stencil_info
+
+        if stencil_info is not None:
+            stencil_info["call_start_time"] = time.perf_counter()
+
```

Comment:
I think this code can be simplified and at the same time keep compatibility with the current `exec_info` usage by doing something like this:

```suggestion
        if exec_info is not None:
            stencil_info = exec_info.setdefault("{{ class_name }}", {})
            exec_info["call_start_time"] = stencil_info["call_start_time"] = time.perf_counter()
```